### PR TITLE
Migrate Transaction

### PR DESCRIPTION
**Pull request by whoojemaflip**
Adds the TransactionPresenter for migrating Transaction content to the new publishing-platform.

The second commit contains a solution to the Downtime problem, specifically working around the previous implementation's reliance on content-api logic and the shared mongo database.  We had to carefully engineer around the publishing-api's workflow - see the commit message for more info.

[Trello](https://trello.com/c/B0WRTEfM/674-migrate-transaction-format-to-rendered)

### CODE REVIEW

File: app/presenters/formats/transaction_presenter.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      {
+        introductory_paragraph: govspeak(:introduction),
+        will_continue_on: edition.will_continue_on,
+        transaction_start_link: edition.link,
+        more_information: govspeak(:more_information),
+        other_ways_to_apply: govspeak(:alternate_methods),
+        what_you_need_to_know: govspeak(:need_to_know),
+        external_related_links: external_related_links,
+        department_analytics_profile: edition.department_analytics_profile
+      }.delete_if { |_, value| value.nil? }
```
Comment (cbaines) [inline app/presenters/formats/transaction_presenter.rb]: :+1: for conciseness

File: test/integration/downtime_test.rb [hunk 2]
```diff
@@ -1,62 +1,113 @@
...
-  teardown do
-    Sidekiq::ScheduledSet.new.clear
+  setup do
+    setup_users
+
+    @edition = FactoryGirl.create(
+      :transaction_edition,
+      :published,
+      title: 'Apply to become a driving instructor',
+      slug: 'foo'
```
Comment (issyl0) [inline test/integration/downtime_test.rb]: Could we make this the slugified version of the title?

Comment (binaryberry) [inline test/integration/downtime_test.rb]: Thanks, addressed now!

File: app/presenters/formats/edition_format_presenter.rb [hunk 3]
```diff
@@ -62,7 +60,7 @@ def details
     end
 
     def external_related_links
-      edition.artefact.external_links.map do |link|
```
Comment (binaryberry) [inline app/presenters/formats/edition_format_presenter.rb]: It doesn't really - extracted to a separate commit: 2c7ed93f.

File: app/services/publishing_api_workflow_bypass_publisher.rb [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    live_edition.present?
+  end
+
+  def discard_draft
+    Services.publishing_api.discard_draft(artefact.content_id)
+  end
+
+  def republish_currently_live_edition
+    PublishingAPIUpdater.new.perform(live_edition.id)
+    PublishingAPIPublisher.new.perform(live_edition.id)
```
Comment (davidbasalla) [inline app/services/publishing_api_workflow_bypass_publisher.rb]: Should this be `PublishingAPIPublisher.new.perform(edition_id, 'republish')` instead? Really not sure on this but wondering if a 'normal' publish would trigger some sort of commpunication/emails/fanfares somewhere?

Comment (whoojemaflip) [inline app/services/publishing_api_workflow_bypass_publisher.rb]: it defaults to 'minor' which shouldn't trigger notifications, but you are right I think this should be 'republish'.
